SPEC.md

```diff
@@ -437,17 +441,19 @@ curl $(echo "$urltmpl" | sed -e "s/{name}/$appname/" -e "s/{version}/$version/"
 
 where _appname_, _version_, _os_, and _arch_ are set to their respective values for the application, and _ext_ is either `aci` or `aci.asc` for retrieving an App Container Image or signature respectively.
 
-In our example above this would be:
+Note that multiple `ac-discovery` tags MAY be returned for a given prefix-match (for example, with different scheme names representing different transport mechanisms).
+In this case, the client implementation MAY choose which to use at its own discretion.
+Public discovery implementations SHOULD always provide at least one HTTPS URL template.
+
+In our example above, using the HTTPS URL template, the client would attempt to retrieve the following URLs:
 
 ```
-sig: https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci.asc
-aci: https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci
-keys: https://example.com/pubkeys.gpg
+Signature: 	https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci.asc
+ACI: 		https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci
+Keys: 		https://example.com/pubkeys.gpg
 ```
 
-This mechanism is only used for discovery of contents URLs.
-
-If the first attempt at fetching the discovery URL returns a status code other than `200 OK`, `3xx`, or does not contain any `ac-discovery` meta tags then the next higher path in the name should be tried.
+If the first attempt at fetching the initial discovery URL returns a status code other than `200 OK`, `3xx`, or does not contain any `ac-discovery` meta tags then the next higher path in the name should be tried.
```

Comment:
I'm not sure if 5xx errors should cause the discover to bubble up. It's usually a temporary error indicative of the server malfunction and should be retried at a later point in time.